python_modules/dagster/dagster_tests/asset_defs_tests/test_asset_deps.py

```diff
@@ -0,0 +1,245 @@
+
+    @asset(deps=["asset_1", "asset_2"])
+    def depends_on_both_sub_assets():
+        return None
+
+    assert len(depends_on_both_sub_assets.input_names) == 2
+    assert depends_on_both_sub_assets.op.ins["asset_1"].dagster_type.kind == DagsterTypeKind.NOTHING
+    assert depends_on_both_sub_assets.op.ins["asset_2"].dagster_type.kind == DagsterTypeKind.NOTHING
+
+    res = materialize(
+        [a_multi_asset, depends_on_one_sub_asset, depends_on_both_sub_assets],
+        resources={"io_manager": TestingIOManager()},
+    )
+    assert res.success
+
+
+def test_mulit_asset_deps_via_key():
+    @multi_asset(
+        outs={
+            "asset_1": AssetOut(),
+            "asset_2": AssetOut(),
+        }
+    )
+    def a_multi_asset():
+        return None, None
+
+    @asset(deps=[AssetKey("asset_1")])
+    def depends_on_one_sub_asset():
+        return None
+
+    assert len(depends_on_one_sub_asset.input_names) == 1
+    assert depends_on_one_sub_asset.op.ins["asset_1"].dagster_type.kind == DagsterTypeKind.NOTHING
+
+    @asset(deps=[AssetKey("asset_1"), AssetKey("asset_2")])
+    def depends_on_both_sub_assets():
+        return None
+
+    assert len(depends_on_both_sub_assets.input_names) == 2
+    assert depends_on_both_sub_assets.op.ins["asset_1"].dagster_type.kind == DagsterTypeKind.NOTHING
+    assert depends_on_both_sub_assets.op.ins["asset_2"].dagster_type.kind == DagsterTypeKind.NOTHING
+
+    res = materialize(
+        [a_multi_asset, depends_on_one_sub_asset, depends_on_both_sub_assets],
+        resources={"io_manager": TestingIOManager()},
+    )
+    assert res.success
+
+
+def test_mulit_asset_deps_via_mixed_types():
+    @multi_asset(
+        outs={
+            "asset_1": AssetOut(),
+            "asset_2": AssetOut(),
+        }
+    )
+    def a_multi_asset():
+        return None, None
+
+    @asset(deps=[AssetKey("asset_1"), "asset_2"])
+    def depends_on_both_sub_assets():
+        return None
+
+    assert len(depends_on_both_sub_assets.input_names) == 2
+    assert depends_on_both_sub_assets.op.ins["asset_1"].dagster_type.kind == DagsterTypeKind.NOTHING
+    assert depends_on_both_sub_assets.op.ins["asset_2"].dagster_type.kind == DagsterTypeKind.NOTHING
+
+    res = materialize(
+        [a_multi_asset, depends_on_both_sub_assets], resources={"io_manager": TestingIOManager()}
+    )
+    assert res.success
+
+
+def test_mulit_asset_deps_via_assets_definition_fails():
+    @multi_asset(
+        outs={
+            "asset_1": AssetOut(),
+            "asset_2": AssetOut(),
+        }
+    )
+    def a_multi_asset():
+        return None, None
+
+    with pytest.raises(
+        CheckError,
+        match="Tried to retrieve asset key from an assets definition with multiple asset keys",
+    ):
+
+        @asset(deps=[a_multi_asset])
+        def depends_on_both_sub_assets():
+            return None
+
+
+def test_source_asset_deps_via_assets_definition():
```

Comment:
This test fails - in order for this to work I think we need to update the `deps` type signature to 
```python
deps: Optional[Sequence[Union[AssetsDefinition, SourceAsset, CoercibleToAssetKey]]]
```
and then handle SourceAssets within the `_make_asset_keys` function

what do you think @alangenfeld  @schrockn ?

Reply:
Yup makes sense